File: 19 - Add Numbers in a Matrix Diagonal/solution.js
function sumMainDiagonal(matrix) {
  // Your code here
  let diagonalSum = 0;
  for (let i = 0; i < matrix.length; i++) {
    diagonalSum += matrix[i][i];
  }

  return diagonalSum;
}

//   Test cases
console.log(
  sumMainDiagonal([
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9],
  ])
);

File: 19 - Add Numbers in a Matrix Diagonal/solution.py
def sum_main_diagonal(matrix):
    # Your code here
    diagonal_sum = 0
    for row in range(len(matrix)):
        diagonal_sum += matrix[row][row]

    return diagonal_sum

#   Test cases    
print(sum_main_diagonal([[1, 2, 3], [4, 5, 6], [7, 8, 9]]))
